### PR TITLE
[93] Handle ActionUsage and ActionDefinition in General View

### PR DESCRIPTION
**Pull request by jerome-obeo**
Adding Action Usage and Action Definition in the ActionFlow section.

Bug: https://github.com/eclipse-syson/syson/issues/93

### CODE REVIEW

File: backend/views/syson-diagram-general-view/src/main/java/org/eclipse/syson/diagram/general/view/GeneralViewDiagramDescriptionProvider.java [hunk 1]
```diff
@@ -71,20 +72,23 @@ public class GeneralViewDiagramDescriptionProvider implements IRepresentationDes
...
     public static  final List<EClass> USAGES = List.of(
+            SysmlPackage.eINSTANCE.getActionUsage(),
             SysmlPackage.eINSTANCE.getAttributeUsage(),
             SysmlPackage.eINSTANCE.getInterfaceUsage(),
             SysmlPackage.eINSTANCE.getItemUsage(),
             SysmlPackage.eINSTANCE.getPartUsage(),
             SysmlPackage.eINSTANCE.getPortUsage());
 
     public static  final Map<EClass, List<EReference>> COMPARTMENTS_WITH_LIST_ITEMS = Map.ofEntries(
+            Map.entry(SysmlPackage.eINSTANCE.getActionDefinition(),      List.of(SysmlPackage.eINSTANCE.getDefinition_OwnedAction())),
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-general-view/src/main/java/org/eclipse/syson/diagram/general/view/GeneralViewDiagramDescriptionProvider.java:83]: Could you also add ItemUsage compartment on ActionDefinition please? It seems often used.

**Review by AxelRICHARD — CHANGES_REQUESTED** [review]
Could you rename the commit with something like:
[93] Handle ActionUsage and ActionDefinition in General View
please?

File: backend/views/syson-diagram-general-view/src/main/java/org/eclipse/syson/diagram/general/view/GeneralViewDiagramDescriptionProvider.java (hunk 1 above)
Comment (jerome-obeo) [reply backend/views/syson-diagram-general-view/src/main/java/org/eclipse/syson/diagram/general/view/GeneralViewDiagramDescriptionProvider.java:83]: Done.

**Comment by jerome-obeo** [comment]
> Could you rename the commit with something like: [93] Handle ActionUsage and ActionDefinition in General View please?

Done.

File: backend/services/syson-services/src/main/java/org/eclipse/syson/util/IDescriptionNameGenerator.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import org.eclipse.emf.ecore.EClassifier;
+import org.eclipse.emf.ecore.EReference;
+import org.eclipse.sirius.components.diagrams.description.NodeDescription;
+
+/**
+ * Interface of name generator used by all SysON description providers.
+ *
+ * @author Jerome Gout
+ */
+public interface IDescriptionNameGenerator {
```
Comment (AxelRICHARD) [inline backend/services/syson-services/src/main/java/org/eclipse/syson/util/IDescriptionNameGenerator.java:25]: Thank you for the comments on all methods.
I'm not sure that all the `@return` comments are appropriate though.
The interface does not have to tell what should be return. It is the implementation responsibility.
What do you think about?

File: backend/views/syson-diagram-common-view/.checkstyle [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<fileset-config file-format-version="1.2.0" simple-config="false" sync-formatter="false">
+  <fileset name="all" enabled="true" check-config-name="SysON" local="false">
+    <file-match-pattern match-pattern="." include-pattern="true"/>
+  </fileset>
+</fileset-config>
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-common-view/.checkstyle:7]: Could you add this
```
<filter name="FilesFromPackage" enabled="true">
    <filter-data value="src/main/resources"/>
  </filter>
```
section please?

File: backend/views/syson-diagram-general-view/src/main/java/org/eclipse/syson/diagram/general/view/GeneralViewDescriptionProvider.java [hunk 4]
```diff
@@ -40,49 +26,19 @@
...
-public class GeneralViewDescriptionProvider implements IEditingContextRepresentationDescriptionProvider {
-
-    private static final String GENERAL_VIEW_DIAGRAM_ID = "GeneralViewDiagram";
-
-    private final IViewConverter viewConverter;
-
-    private final EPackage.Registry ePackagesRegistry;
-
-    private final IInMemoryViewRegistry inMemoryViewRegistry;
+public class GeneralViewDescriptionProvider extends AbstractViewDescriptionProvider implements IEditingContextRepresentationDescriptionProvider {
```
Comment (AxelRICHARD) [inline backend/views/syson-diagram-general-view/src/main/java/org/eclipse/syson/diagram/general/view/GeneralViewDescriptionProvider.java:29]: Could you remove `implements IEditingContextRepresentationDescriptionProvider` please as it is already handled by `AbstractViewDescriptionProvider`?
Could you also remove from every other class inheriting `AbstractViewDescriptionProvider`.

**Review by AxelRICHARD — COMMENTED** [review]
Thank you for your work on this PR, it will increase the quality of the code.
Just a couple minor things to change and it will be ok.

File: backend/application/syson-application/pom.xml [hunk 5]
```diff
@@ -211,6 +211,16 @@
 			<artifactId>syson-diagram-interconnection-view</artifactId>
 			<version>2024.1.0</version>
 		</dependency>
+		<dependency>
```
[the remark this answers is not in the document]
Comment (jerome-obeo) [reply backend/application/syson-application/pom.xml:214]: Done

File: backend/services/syson-services/src/main/java/org/eclipse/syson/util/IDescriptionNameGenerator.java (hunk 2 above)
Comment (jerome-obeo) [reply backend/services/syson-services/src/main/java/org/eclipse/syson/util/IDescriptionNameGenerator.java:25]: Yes, you are right, I introduced implementation details here which is not the right place. 
Such explanations should rather be in the implementation class `DescriptionNameGenerator`

File: backend/views/syson-diagram-common-view/.checkstyle (hunk 3 above)
Comment (jerome-obeo) [reply backend/views/syson-diagram-common-view/.checkstyle:7]: Done

File: backend/views/syson-diagram-general-view/src/main/java/org/eclipse/syson/diagram/general/view/GeneralViewDescriptionProvider.java (hunk 4 above)
Comment (jerome-obeo) [reply backend/views/syson-diagram-general-view/src/main/java/org/eclipse/syson/diagram/general/view/GeneralViewDescriptionProvider.java:29]: I thought that Spring boot required explicitly this implements directive to inject properly the list of `IEditingContextRepresentationDescriptionProvider` in `EditingContextSearchService`
I will remove this.

Comment (jerome-obeo) [reply backend/views/syson-diagram-general-view/src/main/java/org/eclipse/syson/diagram/general/view/GeneralViewDescriptionProvider.java:29]: Done

File: backend/application/syson-application/pom.xml
[the remark this answers is not in the document]
Comment (jerome-obeo) [reply backend/application/syson-application/pom.xml:1]: Done

**Review by AxelRICHARD — APPROVED** [review]

**Pull request merged**